src/image/Image.hpp

```diff
@@ -26,32 +27,48 @@ class Result
 class Image
 {
 public:
+  enum Format { Byte, Float };
+
+  static Image makeEmpty();
+
   static Result<Image> loadPFM(std::string const& path);
   static Result<Image> loadPIC(std::string const& path);
   static Result<Image> loadEXR(std::string const& path);
+  static Result<Image> loadImage(std::string const& path);
 
   int width() const { return width_; }
   int height() const { return height_; }
   int channels() const { return channels_; }
-  int sizeInBytes() const { return width_ * height_ * channels_ * sizeof(float); }
-  float const* data() const { return data_.get(); }
+  int pixelSizeInBytes() const { return format_ == Byte ? sizeof(uint8_t) : sizeof(float); }
+  int sizeInBytes() const { return width_ * height_ * channels_ * pixelSizeInBytes(); }
+  Format format() const { return format_; }
+  void const* data() const { return data_.data(); }
 
-  float value(int x, int y, int channel) const
+  template<typename T>
```

Comment:
Not yet. I want to use it in ImageDocument::pixelValue() to retrieve pixel values as float and byte.